src/Nethermind/Nethermind.Blockchain/BlockProcessor.cs

```diff
@@ -336,5 +367,31 @@ private void ApplyDaoTransition()
                 _stateProvider.SubtractFromBalance(daoAccount, balance, Dao.Instance);
             }
         }
+        
+        private class CompositeAdditionalBlockProcessor : IAdditionalBlockProcessor
+        {
```

Comment:
not moving it one level app?
notice the interface on the top level is simple:

Reply:
Block[] Process(Keccak branchStateRoot, Block[] suggestedBlocks, ProcessingOptions processingOptions, IBlockTracer blockTracer);
        event EventHandler<BlockProcessedEventArgs> BlockProcessed;
        event EventHandler<TxProcessedEventArgs> TransactionProcessed;